### ISSUE TITLE
thread 'main' panicked at 'range end out of bounds'

### ISSUE THREAD

**Issue by nirokato** [maxcountryman/warc-parquet#2]
**Issue Title**: `thread 'main' panicked at 'range end out of bound`

---

### 🐛 Description

- **What did you do?**
I attempted to convert a Common Crawl WARC file to a Parquet file for analysis in DuckDB. I downloaded the file, attempted to process it with `warc-parquet`, but it wasn't able to correctly/fully process the warc.gz file.

`export RUST_BACKTRACE=full; cat CC-NEWS-20230803071746-00285.warc.gz | gzip -d | warc-parquet > CC-NEWS-20230803071746-00285.warc.gz.parquet`

- **What did you expect to see?**
  A new parquet file with a similar size to the original compressed warc.gz file, around 1 GiB.

- **What did you see instead?**
  A new parquet file with only 6,476,466 bytes.

---

### 📎 Steps to Reproduce

```
wget https://data.commoncrawl.org/crawl-data/CC-NEWS/2023/08/CC-NEWS-20230803071746-00285.warc.gz
# Length: 1072778533 (1023M) [binary/octet-stream]
# Saving to: ‘CC-NEWS-20230803071746-00285.warc.gz’
export RUST_BACKTRACE=full; cat CC-NEWS-20230803071746-00285.warc.gz | gzip -d | warc-parquet > CC-NEWS-20230803071746-00285.warc.gz.parquet
# thread 'main' panicked at 'range end out of bounds: 18446744071562881016 <= 6014243295', /home/niro/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bytes-1.4.0/src/bytes.rs:261:9
```

---

### 🖥 System Information

- **Rust Version:** `rustc 1.72.0 (5680fa18f 2023-08-23)` 
- **Operating System and Version:** Ubuntu 22.04.3 LTS
- **warc-parquet version:** warc-parquet 0.4.0

---

### 📝 Additional Context

- **Logs, error messages, or panic messages:** See attached backtrace.txt
[backtrace.txt](https://github.com/maxcountryman/warc-parquet/files/12446627/backtrace.txt)

**Comment by maxcountryman** [comment]
Thanks for reporting this and providing the repro steps. I'll have a look this afternoon.

**Comment by maxcountryman** [comment]
This should be addressed by the latest commit.

At least on my machine, I was able to process the example WARC in about seven seconds and get back valid-looking Parquet on the other end of that.

```txt
D select type, id from 'CC-NEWS-20230803071746-00285.warc.gz.parquet';
┌──────────┬─────────────────────────────────────────────────┐
│   type   │                       id                        │
│ varchar  │                     varchar                     │
├──────────┼─────────────────────────────────────────────────┤
│ warcinfo │ <urn:uuid:3ef2a5a5-d78f-412b-a76a-327e7df13d16> │
│ request  │ <urn:uuid:ce5f6bc3-6052-4eee-be65-1339823b0419> │
│ response │ <urn:uuid:166cecaa-0ceb-4d2a-8332-2ee1dda35eb6> │
│ request  │ <urn:uuid:59d437d0-e130-4378-843c-fecd0e7e6c32> │
│ response │ <urn:uuid:bf4a2657-05f6-4b3b-b4d2-7097afb792b8> │
│ request  │ <urn:uuid:8b6898ef-9050-4424-923f-0122b0c46e86> │
│ response │ <urn:uuid:1213bae8-f7f4-4f68-bf85-382c6a27434a> │
│ request  │ <urn:uuid:7014a6f2-6b97-4274-be73-08e883e3588d> │
│ response │ <urn:uuid:659b7b49-3a47-432e-b997-50da192019d4> │
│ request  │ <urn:uuid:a3042516-1699-4dc4-aa32-015f487f5c08> │
│ response │ <urn:uuid:cc4f461c-053a-48c7-b16f-4e0357e1fc72> │
│ request  │ <urn:uuid:34805f92-a9f7-4e35-a7cf-a38dde89ab6d> │
│ response │ <urn:uuid:eb37be47-4c70-43af-b744-4a02cc180435> │
│ request  │ <urn:uuid:1c8937f6-565a-46f9-b5fe-1c12a908297e> │
│ response │ <urn:uuid:b459cf20-16e5-408a-970e-51f8aa1baaaa> │
│ request  │ <urn:uuid:537721f5-5576-4d87-accf-8054b291275f> │
│ response │ <urn:uuid:65de8f8b-8bfd-4d11-97e9-d2127e24bdbd> │
│ request  │ <urn:uuid:c0c4fbf7-2eb3-474d-accf-328d264664ec> │
│ response │ <urn:uuid:2727f64f-67e2-4971-975f-82ab8811fd4f> │
│ request  │ <urn:uuid:54c27194-84ff-4918-93e5-09a7e0cb1298> │
│    ·     │                        ·                        │
│    ·     │                        ·                        │
│    ·     │                        ·                        │
│ request  │ <urn:uuid:2a024b3e-edd5-41db-ad17-0f79fa3f21f0> │
│ response │ <urn:uuid:1b0ff88e-d90e-4798-bcee-528ed611b86c> │
│ request  │ <urn:uuid:6df46512-0526-41b5-b7ad-83eb9dfa022f> │
│ response │ <urn:uuid:d28d0720-fd96-4c9e-9dc6-0ee1acc67077> │
│ request  │ <urn:uuid:789aeb93-9041-4b98-b907-5b34772e862c> │
│ response │ <urn:uuid:806be384-8358-43cf-b9cd-a1e09dd18333> │
│ request  │ <urn:uuid:37760c19-a767-4372-8d1c-938c771eebc3> │
│ response │ <urn:uuid:497f6191-0cb2-4fbd-8d36-45b4f0c424d4> │
│ request  │ <urn:uuid:da224d04-1490-496b-bcda-d3af99f85646> │
│ response │ <urn:uuid:b827b931-0dc5-4d3d-8456-d0ffd788a644> │
│ request  │ <urn:uuid:f7e547c9-13eb-4816-8bc1-a96fb9f1e211> │
│ response │ <urn:uuid:ec30223f-0a91-4b0d-a1e4-dd4e19c71c40> │
│ request  │ <urn:uuid:e6d5b3b4-62dd-45f8-9466-b2f71b9f025c> │
│ response │ <urn:uuid:2d316653-be8f-4481-b54d-cc7b09b0ff79> │
│ request  │ <urn:uuid:04fb6a69-fe76-4d51-b747-7c646a20abf5> │
│ response │ <urn:uuid:6cd16e91-1ca5-49b1-8a52-72510405ee66> │
│ request  │ <urn:uuid:61e37130-845b-4135-a597-82317c797eae> │
│ response │ <urn:uuid:d5ff4719-17b6-4afe-bb60-997d4dc24ee2> │
│ request  │ <urn:uuid:0270ef8f-092c-4786-b6bd-36c350d025a2> │
│ response │ <urn:uuid:59c0aa38-a708-48f8-b155-64dc7e7560e8> │
├──────────┴─────────────────────────────────────────────────┤
│ 48537 rows (40 shown)                            2 columns │
└────────────────────────────────────────────────────────────┘
```

If this looks good to you I'm happy to publish a new release.

**Comment by nirokato** [comment]
Yes, confirmed working on my end with multiple warc files. Please publish the release for others. Thank you for the superb turn around time!

**Comment by maxcountryman** [comment]
Great to hear. I'll publish a new release in the morning.

**Comment by maxcountryman** [comment]
I released `0.5.0` this morning.

Thank you again for finding and reporting this!
